--- staging_dir/host/bin/libtoolize	2014-09-21 04:49:01.000000000 +0900
+++ staging_dir/host/bin/libtoolize.new	2016-07-07 16:01:40.000000000 +0900
@@ -326,9 +326,21 @@

 : ${CP="cp -f"}
 test "${ECHO+set}" = set || ECHO=${as_echo-'printf %s\n'}
-: ${EGREP="/BB/build/ar71xx/generic/staging_dir/host/bin/grep -E"}
-: ${FGREP="/BB/build/ar71xx/generic/staging_dir/host/bin/grep -F"}
-: ${GREP="/BB/build/ar71xx/generic/staging_dir/host/bin/grep"}
+if test -n "$STAGING_DIR"; then
+	: ${EGREP="$STAGING_DIR/../host/bin/grep -E"}
+else
+	: ${EGREP="/BB/build/ar71xx/generic/staging_dir/host/bin/grep -E"}
+fi
+if test -n "$STAGING_DIR"; then
+	: ${FGREP="$STAGING_DIR/../host/bin/grep -F"}
+else
+	: ${FGREP="/BB/build/ar71xx/generic/staging_dir/host/bin/grep -F"}
+fi
+if test -n "$STAGING_DIR"; then
+	: ${GREP="$STAGING_DIR/../host/bin/grep"}
+else
+	: ${GREP="/BB/build/ar71xx/generic/staging_dir/host/bin/grep"}
+fi
 : ${LN_S="ln -s"}
 : ${MAKE="make"}
 : ${MKDIR="mkdir"}
