--- a/src/system.h
+++ b/src/system.h
@@ -171,8 +171,10 @@
 #endif
 
 #ifdef HAVE_NET_ETHERNET_H
+#ifdef __UCLIBC__
 #include <net/ethernet.h>
 #endif
+#endif
 
 #ifdef HAVE_ASM_TYPES_H
 #include <asm/types.h>
--- a/src/chilli_limits.h
+++ b/src/chilli_limits.h
@@ -18,8 +18,8 @@
  * 
  */
 
-#ifndef _LIMITS_H
-#define _LIMITS_H
+#ifndef _CHILLI_LIMITS_H
+#define _CHILLI_LIMITS_H
 
 /*
  * extracted from various .h files, needs some cleanup.
