--- a/src/linux/xt_coova.c
+++ b/src/linux/xt_coova.c
@@ -305,14 +305,18 @@ static int coova_mt_check(const struct x
 	    strnlen(info->name, XT_COOVA_NAME_LEN) == XT_COOVA_NAME_LEN)
 		return -EINVAL;
 
+#ifdef CONFIG_DEBUG2
 	printk(KERN_INFO "xt_coova: looking for %s\n", info->name);
+#endif
 
 	mutex_lock(&coova_mutex);
 	t = coova_table_lookup(info->name);
 	if (t != NULL) {
 		t->refcnt++;
+#ifdef CONFIG_DEBUG2
 		printk(KERN_INFO "xt_coova: found %s refcnt=%d\n", 
 		       info->name, t->refcnt);
+#endif
 		goto out;
 	}
 
@@ -347,11 +351,15 @@ static int coova_mt_check(const struct x
 	spin_lock_bh(&coova_lock);
 	list_add_tail(&t->list, &tables);
 	spin_unlock_bh(&coova_lock);
+#ifdef CONFIG_DEBUG2
 	printk(KERN_INFO "xt_coova: created %s refcnt=%d\n", 
 	       t->name, t->refcnt);
+#endif
 out:
 	mutex_unlock(&coova_mutex);
+#ifdef CONFIG_DEBUG2
 	printk(KERN_INFO "xt_coova: match ret=%d\n", ret); 
+#endif
 	return ret;
 }
 
