--- a/conf/up.sh.in
+++ b/conf/up.sh.in
@@ -28,7 +28,7 @@ run_up() {
 	    ipt -I FORWARD -o $DHCPIF -m coova --name $KNAME --dest -j ACCEPT
 	    ipt -I FORWARD -i $TUNTAP -j ACCEPT
 	    ipt -I FORWARD -o $TUNTAP -j ACCEPT
-	    [ -n "$DHCPLISTEN" ] && ifconfig $DHCPIF $DHCPLISTEN
+	    [ -n "$DHCPLISTEN" ] && ifconfig $DHCPIF $DHCPLISTEN netmask $MASK
 	else
 	    if [ "$LAYER3" != "1" ]
 	    then
