--- a/src/dhcp.c
+++ b/src/dhcp.c
@@ -864,7 +864,7 @@ int nfqueue_cb_in(struct nfq_q_handle *q
   struct nfqnl_msg_packet_hdr *ph;
   struct nfqnl_msg_packet_hw *hw;
   u_int32_t id = 0;
-  char *data;
+  unsigned char *data;
   int ret;
 
   int result = NF_DROP;
@@ -927,7 +927,7 @@ int nfqueue_cb_out(struct nfq_q_handle *
 		   struct nfq_data *nfa, void *cbdata) {
   struct nfqnl_msg_packet_hdr *ph;
   u_int32_t id = 0;
-  char *data;
+  unsigned char *data;
   int ret;
 
   int result = NF_DROP;
--- a/src/queue.c
+++ b/src/queue.c
@@ -13,7 +13,7 @@ static u_int32_t print_pkt (struct nfq_d
   struct nfqnl_msg_packet_hdr *ph;
   u_int32_t mark,ifi; 
   int id = 0;
-  char *data;
+  unsigned char *data;
   int ret;
   
   ph = nfq_get_msg_packet_hdr(tb);
