From 08311fb8f8fbe397398e42a82ac52d6d0110cc70 Mon Sep 17 00:00:00 2001
From: Baligh GUESMI <gasmibal@gmail.com>
Date: Mon, 23 Nov 2015 12:33:17 +0100
Subject: [PATCH] close socket before return on bind() failure.

---
 src/rtmon.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/src/rtmon.c
+++ b/src/rtmon.c
@@ -57,8 +57,11 @@ static int open_netlink() {
   addr.nl_pid = getpid();
   addr.nl_groups = MYMGRP;
 
-  if (bind(sock,(struct sockaddr *)&addr,sizeof(addr)) < 0)
+  if (bind(sock,(struct sockaddr *)&addr,sizeof(addr)) < 0) {
+    syslog(LOG_ERR, "Binding on socket(%p) failed.", sock);
+    close(sock);
     return -1;
+  }
 
   return sock;
 }
