--- a/src/dhcp.c
+++ b/src/dhcp.c
@@ -412,7 +412,9 @@ int dhcp_net_send(struct _net_interface 
   syslog(LOG_DEBUG, "dhcp_send() len=%d", length);
 #endif
 
-  return net_write_eth(netif, packet, length, &netif->dest);
+  if (length > 54) {
+    return net_write_eth(netif, packet, length, &netif->dest);
+  }
 
 #elif defined (__FreeBSD__) || defined (__APPLE__) || defined (__OpenBSD__) || defined (__NetBSD__)
   if (safe_write(netif->fd, packet, length) < 0) {
